packages/link/src/__demo__/examples/reactRouter.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright 2017 CA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+import Link from '../../Link';
+import {
+  BrowserRouter,
+  Link as ReactRouterLink,
+  Route,
+  Switch
+} from 'react-router-dom';
+
+export default {
+  title: 'Using React Router',
+  description: `In this example, we import { Link as ReactRouterLink } from 'react-router-dom' so that we can get client-side routing with our styles.`,
+  scope: { Link, ReactRouterLink, BrowserRouter, Switch, Route },
+  source: `<BrowserRouter>
```

Comment:
_Not relevant to this PR_: We thought of the need for examples with a hidden source, but didn't think of examples that need to hide the preview, like this one.